lib/payment/payment_service.dart

```diff
@@ -0,0 +1,154 @@
+import 'package:coffeecard/data/repositories/v2/purchase_repository.dart';
+import 'package:coffeecard/generated/api/coffeecard_api_v2.swagger.swagger.dart';
+import 'package:coffeecard/service_locator.dart';
+import 'package:coffeecard/widgets/popup_card.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+enum InternalPaymentType {
+  mobilePay,
+  applePay,
+}
+
+abstract class PaymentService {
```

Comment:
Maybe PaymentHandler is a more suiting since this is strictly not a service class with the current usage. Also should be in it's separate class.

Suggested paths
`PaymentService` -> `lib\payment`
`MobilePayService` -> \lib\payment\mobilepay\`
and like for other classes